controllers/crd/crd_controller.go

```diff
@@ -0,0 +1,105 @@
+package crd
+
+import (
+	"context"
+	"slices"
+
+	operatorhandler "github.com/operator-framework/operator-lib/handler"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+var (
+	log = logf.Log.WithName("controller_crd")
+)
+
+// RegisterReconciler creates a new Descheduler Reconciler and registers it into manager.
+func RegisterReconciler(mgr manager.Manager, restartCh chan<- struct{}) error {
+	return add(mgr, newReconciler(mgr, restartCh))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, restartCh chan<- struct{}) reconcile.Reconciler {
+
+	r := &ReconcileCRD{
+		client:       mgr.GetClient(),
+		restartCh:    restartCh,
+		eventEmitter: hcoutil.GetEventEmitter(),
+	}
+
+	return r
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+
+	byNamesFilter := func(names []string) func(object client.Object) bool {
+		return func(object client.Object) bool {
+			return slices.Contains(names, object.GetName())
+		}
+	}
+	// look only at descheduler CRD for now
+	byNamesFuncs := predicate.NewPredicateFuncs(byNamesFilter([]string{hcoutil.DeschedulerCRDName}))
```

Comment:
Nit: Not sure about this. We know the name in filter function context, so I don't think we need to pass it as a parameter. Also, we don't need to pass a slice, since we know that there is only one name. 

In addition, I think we don't have to hold a variable for `byNamesFilter`, because it is used only once, just below. WDYT about this change?
```suggestion
	// look only at descheduler CRD for now
	byNamesFuncs := predicate.NewPredicateFuncs(func(object client.Object) bool {
		return names, object.GetName() == hcoutil.DeschedulerCRDName
	})
```

Reply:
I implemented `byNamesFilter` handling strings because I expect that in the future we can be interested also to other CRDs (this is supposed to be a generic CRD controller not something specific just for the Descheduler one, although now it is).
But yes, for sure I can remove the intermediate variable.